internal/data/spicedb.go

```diff
@@ -287,7 +287,17 @@ func (s *SpiceDbRepository) IsBackendAvaliable() error {
 	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
 
-	_, err := s.healthClient.Check(ctx, &grpc_health_v1.HealthCheckRequest{})
+	resp, err := s.healthClient.Check(ctx, &grpc_health_v1.HealthCheckRequest{})
+	if err != nil {
+		return err
+	}
+
+	switch resp.Status {
+	case grpc_health_v1.HealthCheckResponse_NOT_SERVING, grpc_health_v1.HealthCheckResponse_SERVICE_UNKNOWN:
+		return err
```

Comment:
Wouldn't err be nil here? We got past the nil check above, and it hasn't been written to since.

You probably want an explicit error here (like fmt.Errorf()) unless grpc_health_v1 has something relevant you could return instead.

Reply:
ahh you're right, err would just be nil.

Reply:
Added errors when `SERVING` is not returned ("error connecting to backend"). Also added an error so the timeout returns the proper response.